### PR TITLE
docs: fix invalid link in declarative-setup.md

### PR DESCRIPTION
**Pull request by alexmt**
PR fixes broken link in declarative-setup.md and enables docs validation in PRs

### CODE REVIEW

File: .github/workflows/gh-pages.yaml [hunk 1]
```diff
@@ -4,6 +4,9 @@ on:
   push:
     branches:
       - master
+  pull_request:
+    branches:
+      - 'master'
```
Comment (jannfis) [inline .github/workflows/gh-pages.yaml:9]: What's the reason for this?

**Review by jannfis — APPROVED** [review]
LGTM - just confused by one of the changes (see below)

File: .github/workflows/gh-pages.yaml (hunk 1 above)
Comment (alexmt) [reply .github/workflows/gh-pages.yaml:9]: This is to run docs build on every PR. Step `deploy` that pushes build results is executed only on push.

Comment (jannfis) [reply .github/workflows/gh-pages.yaml:9]: I understand, makes sense. Thanks :+1:

**Pull request merged**